CharsetDeltaJob: Workspace was not properly initialized or has already shutdown

I enabled logging output for the ApiTools Application and it sometimes falls with exception:

```
[ERROR] Workspace was not properly initialized or has already shutdown.
java.lang.Exception: Workspace was not properly initialized or has already shutdown.
    at org.eclipse.core.internal.resources.ResourceException.provideStackTrace (ResourceException.java:42)
    at org.eclipse.core.internal.resources.ResourceException.<init> (ResourceException.java:38)
    at org.eclipse.core.internal.resources.Workspace.getWorkManager (Workspace.java:1817)
    at org.eclipse.core.internal.resources.Workspace.endOperation (Workspace.java:1481)
    at org.eclipse.core.internal.resources.CharsetDeltaJob.run (CharsetDeltaJob.java:203)
    at org.eclipse.core.internal.jobs.Worker.run (Worker.java:63)
```

so it seems the `CharsetDeltaJob` should catch and ignore the exception if it can't make sure that it runs on a closing workspace.

--- a/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetDeltaJob.java
+++ b/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetDeltaJob.java
@@ -236,6 +236,14 @@ public class CharsetDeltaJob extends Job implements IContentTypeManager.IContent
 	}
 
 	public void shutdown() {
+		try {
+			// try to prevent execution of this job to avoid prevent "already shutdown.":
+			cancel();
+			// if job is already running wait for it to finish:
+			join();
+		} catch (InterruptedException e) {
+			// ignore
+		}
 		IContentTypeManager contentTypeManager = Platform.getContentTypeManager();
 		//if the service is already gone there is nothing to do
 		if (contentTypeManager != null)

